core/domain/state_domain.py

```diff
@@ -1919,6 +1922,295 @@ def validate(self) -> None:
                 'or zero if not yet specified %s' %
                 self.duration_secs)
 
+def validate_rte_tags(
+    html: str, is_tags_nested_inside_tabs_or_collapsible: bool = False
+) -> None:
+    """Validate all the RTE tags.
+
+    Args:
+        html: str. The RTE content of the state.
+        is_tags_nested_inside_tabs_or_collapsible: bool. True when we
+            validate tags inside `Tabs` or `Collapsible` tag.
+
+    Raises:
+        utils.ValidationError. Raises the validation error in case the
+            content is not valid.
+    """
+    soup = bs4.BeautifulSoup(html, 'html.parser')
+
+    for tag in soup.find_all('oppia-noninteractive-image'):
+        if not tag.has_attr('alt-with-value'):
+            raise utils.ValidationError(
+                'Image tag does not have \'alt-with-value\' attribute.'
+            )
+
+        if not tag.has_attr('filepath-with-value'):
+            raise utils.ValidationError(
+                'Image tag does not have \'filepath-with-value\' attribute.'
+            )
+
+    for tag in soup.find_all('oppia-noninteractive-skillreview'):
+
+        if not tag.has_attr('text-with-value'):
+            raise utils.ValidationError(
+                'SkillReview tag does not have \'text-with-value\' '
+                'attribute.'
+            )
+
```

Comment:
do not put new line between if and else

Reply:
also, you can remove the `else` line. If it enters inside the `if` condition, it will raise an error and not continue anyways. So, you can cut down on some of your indentation and make it easier to read.

Reply:
Done